src/librustc/util/profiling.rs

```diff
@@ -149,103 +170,244 @@ impl SelfProfiler {
     }
 
     pub fn start_activity(&mut self, category: ProfileCategory) {
```

Comment:
Could you mark all the methods between here and `record` with `#[inline]`? (including `record`)

Reply:
Done